tools/hsmtool.c

```diff
@@ -413,5 +501,20 @@ int main(int argc, char *argv[])
 		                       argv[5], argc >= 7 ? argv[6] : NULL);
 	}
 
+	if (streq(method, "generatehsm")) {
+		if ( argc != 3)
+			show_usage(argv[0]);
+
+		char *hsm_secret_path = argv[2];
+
+		/* if hsm_secred already exists we abort the process
+		 * we do not want to lose someone else's funds */
+		struct stat st;
+		if (stat(hsm_secret_path, &st) == 0)
+			errx(ERROR_USAGE, "hsm_secret file already exists");
```

Comment:
Nice sanity check :)

Reply:
I'd echo the path that already exists here also e.g.
```suggestion
		/* if hsm_secred already exists we abort the process
		 * we do not want to lose someone else's funds */
		struct stat st;
		if (stat(hsm_secret_path, &st) == 0)
			errx(ERROR_USAGE, "/path/to/hsm_secret already exists. Aborting.");
```